compositor_pipeline/src/pipeline/decoder.rs

```diff
@@ -169,3 +152,23 @@ pub struct DecodedDataReceiver {
 pub struct AacDecoderOptions {
     pub asc: Option<Bytes>,
 }
+
+struct DecodedAudioInputInfo {
+    decoded_sample_rate: u32,
+}
+
+#[derive(Debug)]
+struct DecodedSamples {
+    samples: Arc<Vec<(i16, i16)>>,
+    start_pts: Duration,
+    sample_rate: u32,
+}
```

Comment:
I separated this type from `InputSamples`, since:
1. It's better to keep the `sample_rate` instead of `end_pts` in this case
2. It provides better type safety, requiring using the resampler before pushing samples to the queue